lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+/**
+ * MSALModule.java functions as a wrapper around MSAL for Android's signing in, signing out, and acquiring tokens functionality.
+ * It is a proof of concept, focusing on single account mode and AAD.
+ */
+
+package com.reactlibrary;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.Callback;
+
+public class MSALModule extends ReactContextBaseJavaModule {
+
+    private final ReactApplicationContext reactContext;
+    private ISingleAccountPublicClientApplication IPublicClientApplication;
+    private IAccount mAccount;
+
+    public MSALModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
+
+        try{
+            //initialize IPublicClientApplication with config file, located in main/java/res/raw/auth_config_single_account.json
+            IPublicClientApplication = PublicClientApplication.createSingleAccountPublicClientApplication(
```

Comment:
I'm a bit out of my depth here, so I may be wrong about this. Given that the "I" prefix in variable names is usually intended to identify interfaces, and this line is assigning an instance of `PublicClientApplication` to `IPublicClientApplication`, maybe renaming the variable to something like `publicClientApplication` would be a good idea.

Also, the variable declaration in line 18 defines the type as `ISingleAccountPublicClientApplication`, which is an interface (to my knowledge, Java interfaces cannot and should not be instantiated). I think you might want to use the `SingleAccountPublicClientApplication` class, which implements `ISingleAccountPublicClientApplication` instead.

Like I said, a little out of my depth here so let me know if I'm off base!

Reply:
This makes sense and I will go ahead and rename the variable. 
For `ISingleAccountPublicClientApplication` in general, for some reason, `createSingleAccountPublicClientApplication` isn't a method of `singleAccountPublicClientApplication` (I tried this first a while ago) but of the interface instead. I modeled a lot of MSALmodule off of the MSAL Android sample app, which also instantiates `ISingleAccountPublicClientApplication` in a similar way. This also confuses me and I should have look into a better  explanation earlier and will do so now!

Reply:
To be honest, that is a very good reason for having the code the way you wrote it. If it works and it matches the Android sample you must be on the right track. Maybe someone from the Android team can shed a little light on why the code is organized/works that way.